socorro/lib/transaction.py

```diff
@@ -117,38 +117,33 @@ def __call__(self, function, *args, **kwargs):
                         last_failure = sys.exc_info()
                         connection.rollback()
                         reraise(*last_failure)
-
-            except self.db_conn_context_source.conditional_exceptions as x:
-                # these exceptions may or may not be retriable the test is
-                # for is a last ditch effort to see if we can retry
-                if not self.db_conn_context_source.is_operational_exception(x):
-                    # If the logger exists, log the issue, otherwise print it
-                    # to stdout.
-                    if self.config.logger:
+            except Exception as x:
+                if type(x) in self.db_conn_context_source.conditional_exceptions:
```

Comment:
This has sublty different behavior than what we had before. I think you need to do:

```
if self.db_conn_context_source.conditional_exceptions and isinstance(x, self.db_conn_context_source.conditional_exceptions):
```

which will correctly handle subclasses of exception classes in `conditional_exceptions`. Same thing for the other type checks.

I don't know offhand if there are cases of subclasses that need to get handled, but I'm not sure I could figure it out without knowing exactly what gets thrown where and Python is a crummy language for that kind of static analysis.

Reply:
Tried this yesterday and it isn't so straightforward. `isinstance` throws a type error for some of the classes. I'll need to dig into it a little.

Reply:
Let me know if you want me to poke at it. I think I'll be switching to Python 3 migration in the next week.

Reply:
Take it if you'd like!

Reply:
Just close this out to let me know you're on it and I'll cease my end